### PR TITLE
chore: add missing utm parameters for cloud links to Debug page

### PR DESCRIPTION
**Pull request by warrensplayer**
Four links on the Debug page were missing UTM parameters needed for measuring impact of certain features
* artifact links
* "View in Cypress Cloud" link in the header
* Link to the project in the run navigation limit banner
* Link to the run in the spec limit banner

### Steps to test

Verify by reviewing the E2E and CT tests

### How has the user experience changed?

The user experience was not affected.

### PR Tasks

- [X] Have tests been added/updated?
- [-] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [-] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

File: packages/app/src/debug/DebugRunNavigationLimitMessage.vue [hunk 1]
```diff
@@ -22,9 +22,12 @@
...
 import { useI18n } from '@cy/i18n'
 import { IconWarningCircle } from '@cypress-design/vue-icon'
+import { getUrlWithParams } from '@packages/frontend-shared/src/utils/getUrlWithParams'
 
 const { t } = useI18n()
 
-defineProps<{ cloudProjectUrl?: string }>()
+const props = defineProps<{ cloudProjectUrl?: string }>()
+
+const cloudProjectUrlWithUtmParams = props.cloudProjectUrl && getUrlWithParams({ url: props.cloudProjectUrl, params: { utm_medium: 'Debug Tab', utm_campaign: 'Run Navigation Limit' } })
```
Comment (mike-plummer) [inline packages/app/src/debug/DebugRunNavigationLimitMessage.vue]: Should this be `computed(() => {})` so it's reactive to prop changes after mount?

File: packages/app/src/debug/DebugSpecLimitBanner.vue [hunk 2]
```diff
@@ -26,14 +26,17 @@
...
 
 const { t } = useI18n()
 
-defineProps<{
+const props = defineProps<{
   failedTestCount: number
   cloudRunUrl: string | null
 }>()
 
+const cloudRunUrlWithUtmParams = props.cloudRunUrl && getUrlWithParams({ url: props.cloudRunUrl, params: { utm_medium: 'Debug Tab', utm_campaign: 'Spec Limit' } })
```
Comment (mike-plummer) [inline packages/app/src/debug/DebugSpecLimitBanner.vue]: computed?

File: packages/app/src/debug/DebugPageHeader.vue [hunk 3]
```diff
@@ -38,7 +38,7 @@
         <li class="font-normal text-sm text-indigo-500 inline">
           <ExternalLink
             data-cy="debug-header-dashboard-link"
-            :href="debug.url || '#'"
+            :href="runUrl || '#'"
```
Comment (mike-plummer) [inline packages/app/src/debug/DebugPageHeader.vue]: Total nitpick - you could encapsulate the `|| '#'` case into an `if/else` conditional in the `computed` block

File: packages/app/src/debug/utils/debugArtifacts.ts [hunk 4]
```diff
@@ -1,23 +1,28 @@
 import type { CloudRunInstance } from '@packages/data-context/src/gen/graphcache-config.gen'
+import { getUrlWithParams } from '@packages/frontend-shared/src/utils/getUrlWithParams'
 import type { useI18n } from '@cy/i18n'
 
 export type ArtifactType = 'TERMINAL_LOG' | 'IMAGE_SCREENSHOT' | 'PLAY'
 
 export type DebugArtifact = { icon: ArtifactType, text: string, url: string }
 
+const formatUrl = (url: string, campaign: string): string => {
+  return getUrlWithParams({ url, params: { utm_medium: 'Debug Tag', utm_campaign: campaign } })
```
Comment (astone123) [inline packages/app/src/debug/utils/debugArtifacts.ts]: Should this be `Debug Tab`?

**Review by astone123 — APPROVED** [review]
Looks good, one typo I noticed

File: packages/app/src/debug/utils/debugArtifacts.ts (hunk 4 above)
Comment (warrensplayer) [reply packages/app/src/debug/utils/debugArtifacts.ts]: Yes, moved to constant and refactor usages of this string

File: packages/app/src/debug/DebugPageHeader.vue (hunk 3 above)
Comment (warrensplayer) [reply packages/app/src/debug/DebugPageHeader.vue]: Good call out.  Made the update

File: packages/app/src/debug/DebugSpecLimitBanner.vue (hunk 2 above)
Comment (warrensplayer) [reply packages/app/src/debug/DebugSpecLimitBanner.vue]: Done

File: packages/app/src/debug/DebugRunNavigationLimitMessage.vue (hunk 1 above)
Comment (warrensplayer) [reply packages/app/src/debug/DebugRunNavigationLimitMessage.vue]: Done

**Review by mike-plummer — APPROVED** [review]

**Pull request merged**